core/src/main/kotlin/com/alecstrong/sql/psi/core/SqlCoreEnvironment.kt

```diff
@@ -77,8 +89,10 @@ open class SqlCoreEnvironment(
       ProjectRootManager::class.java,
       ProjectRootManagerImpl(projectEnvironment.project),
```

Comment:
```
Plugin com.squareup.sqldelight:2.1.0-SNAPSHOT-1706283044420 against IC-233.13135.103: 1 compatibility problem. 10 usages of scheduled for removal API and 15 usages of deprecated API. 6 usages of experimental API. 11 usages of internal API
Compatibility problems (1): 
    #Invocation of unresolved constructor com.intellij.openapi.roots.impl.ProjectRootManagerImpl.<init>(Project)
        Constructor com.alecstrong.sql.psi.core.SqlCoreEnvironment.<init>(java.util.List sourceFolders, java.util.List dependencies) contains an *invokespecial* instruction referencing an unresolved constructor com.intellij.openapi.roots.impl.ProjectRootManagerImpl.<init>(com.intellij.openapi.project.Project). This can lead to **NoSuchMethodError** exception at runtime.
```

This is not compatible with 2023.3.3

Reply:
we may want to consider moving the `Environment` types to their own module that doesn't get checked for intellij compatibility - since its only used by the compiler we know what version its on so don't have to worry about forwards compatibility

Reply:
we did the same in sqldelight by moving the environment stuff to the gradle plugin so it wouldn't be checked: https://github.com/cashapp/sqldelight/blob/master/sqldelight-gradle-plugin/src/main/kotlin/app/cash/sqldelight/core/SqlDelightEnvironment.kt

Reply:
So what's next? Should I move it to another module or do we just ignore it?

Reply:
can you move it to another module? maybe `:compiler` and then we just add that as a dependency of the gradle plugin

Reply:
Oh, I named it `environment`. What do you prefer?

Reply:
works for me